File: Modulo08/AlunoEndereco-Java/Endereco.java
public class Endereco {

    private String logradouro;
    private int numero;
    private String cidade;
    private String estado;

    public Endereco(String logradouro, int numero, String cidade, String estado) {
        this.logradouro = logradouro;
        this.numero = numero;
        this.cidade = cidade;
        this.estado = estado;
    }

    public String getLogradouro() {
        return this.logradouro;
    }

    public void setLogradouro(String logradouro) {
        this.logradouro = logradouro;
    }

    public int getNumero() {
        return this.numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getCidade() {
        return this.cidade;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }

    public String getEstado() {
        return this.estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public void imprime() {
        System.out.println("Logradouro: " + this.getLogradouro());
        System.out.println("Numero: " + this.getNumero());
        System.out.println("Cidade: " + this.getCidade());
        System.out.println("Estado: " + this.getEstado());
    }
}

File: Modulo05/AlunoEndereco-Referencia/Endereco.cpp
#include "Endereco.h"
#include <iostream>

using namespace std;

// construtor com 4 parâmetros

Endereco::Endereco(string logradouro, int numero, string cidade, string estado) {

    this->logradouro = logradouro;
    this->numero = numero;
    this->cidade = cidade;
    this->estado = estado;
}

// destrutor da classe

Endereco::~Endereco() {
    cout << "Destrutor Endereco (" << logradouro << ", " << numero << ")" << endl;
}

string Endereco::getLogradouro() const {
    return logradouro;
}

void Endereco::setLogradouro(string val) {
    logradouro = val;
}

int Endereco::getNumero() const {
    return numero;
}

void Endereco::setNumero(int val) {
    numero = val;
}

string Endereco::getCidade() const {
    return cidade;
}

void Endereco::setCidade(string val) {
    cidade = val;
}

string Endereco::getEstado() const {
    return estado;
}

void Endereco::setEstado(string val) {
    estado = val;
}

void Endereco::imprime() const {
    cout << "Logradouro: " << this->getLogradouro() << endl;
    cout << "Numero: " << this->getNumero() << endl;
    cout << "Cidade: " << this->getCidade() << endl;
    cout << "Estado: " << this->getEstado() << endl;
}
